corehq/apps/smsbillables/migrations/0019_deactivate_hardcoded_twilio_gateway_fees.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.12 on 2018-04-17 18:51
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from corehq.messaging.smsbackends.twilio.models import SQLTwilioBackend
+from corehq.sql_db.operations import HqRunPython
+
+
+def deactivate_hardcoded_twilio_gateway_fees(apps, schema_editor):
+    SmsGatewayFeeCriteria = apps.get_model('smsbillables', 'SmsGatewayFeeCriteria')
+
+    to_deactivate = SmsGatewayFeeCriteria.objects.filter(
+        backend_api_id=SQLTwilioBackend.get_api_id(),
+        country_code__isnull=False,
+        is_active=True,
+    )
+    remaining = SmsGatewayFeeCriteria.objects.filter(
+        backend_api_id=SQLTwilioBackend.get_api_id(),
+    ).exclude(
+        id__in=to_deactivate.values('id')
```

Comment:
should this be `values_list('id', flat=True)`?

Reply:
I think `values_list('id', flat=True)` will result in two queries, the first to get the ids and put them in a list (in Python), and the second like
```sql
SELECT ... WHERE id IN ?
```
`?` is replaced with the list of ids.

On the other hand `values('id')` will make a single query like
```sql
SELECT ... WHERE id IN (SELECT id FROM ...)
```
the second select statement is the SQL of the `values('id')` queryset.

Reply:
i see, thanks for the explanation